sortinghat/core/schema.py

```diff
@@ -77,6 +82,61 @@ def convert_json_field_to_generic_scalar(field, registry=None):
     return OperationArgsType(description=field.help_text, required=not field.null)
 
 
+def parse_date_filter(filter_value):
+    """Extract the filter terms from a date filter
+
+    The accepted formats are controlled by regular expressions
+    matching two patterns: a comparison operator (>, >=, <, <=) and a date
+    OR a range operator (..) between two dates.
+
+    The accepted date format is ISO 8601, YYYY-MM-DDTHH:MM:SS, also
+    accepting microseconds and time zone offset (YYYY-MM-DDTHH:MM:SS.ms+HH:HH).
+
+    :param filter_value: String containing the filter value
+
+    :returns: A dictionary including an operator and the date values
+    """
+    # Accepted date format is ISO 8601, YYYY-MM-DDTHH:MM:SS
+    filter_data = {
+        "operator": None,
+        "date1": None,
+        "date2": None
+    }
+
+    iso_date_group = "(\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}(\.\d+)?(\+\d{2}:\d{2})?)"
+    # Filter with a comparison operator (>, >=, <, <=) and a date (e.g. `>=YYYY-MM-DDTHH:MM:SS`)
+    oper_comparison = r"^(<=?|>=?)%s$" % iso_date_group
+    # Filter with a range operator (..) between two dates
+    # (e.g. YYYY-MM-DDTHH:MM:SS..YYYY-MM-DDTHH:MM:SS)
+    range_comparison = r"^%s\.{2}%s$" % (iso_date_group, iso_date_group)
+
+    oper_result = re.match(oper_comparison, filter_value)
+    range_result = re.match(range_comparison, filter_value)
+
+    if not oper_result and not range_result:
+        raise InvalidFilterError(msg='Filter format is not valid')
```

Comment:
It would be better if you explain what the problem is and how someone can fix it.

Reply:
In this case I think it is better to include some basic documentation within the field itself using this graphene structure when the filter is declared:
```
 graphene.String(
        required=False,
        description='Filter with a comparison operator (>, >=, <, <=) and a date OR with a range operator (..) between two dates. E.g. `>=YYYY-MM-DDTHH:MM:SS`.'
    )
```
About the information the error has to provide, the problem there is the filter did not match with any of the accepted regular expressions. Should this error print the not matching parts together with the expression that should match?

Reply:
This description would look like this in the API web documentation:

![doc-last-updated-filter](https://user-images.githubusercontent.com/9032790/100731172-52797000-33cb-11eb-8d26-a337716ac066.png)

@sduenas, WDYT ?

Reply:
I think it's good to have that in the documentation. However, one of the problems of the exception is that it doesn't says what filter is failing, so the user won't be able to know what to change.

Reply:
Oh, I see what you meant. The error was not telling `last_updated` is the filter field that is failing, I've just updated it catching the exception when the filter fails while the query is being resolved.

Reply:
BTW, I kept the piece of documentation I posted above with some real examples:
* `>=2020-10-12T09:35:06.13045+01:00` 
* `2020-10-12T00:00:00..2020-11-22T00:00:00`